plugins/ssr/ssrequality.ml

```diff
@@ -287,7 +287,10 @@ let foldtac occ rdx ft gl =
     (fun env c _ h -> try find_T env c h ~k:(fun env t _ _ -> t) with NoMatch ->c),
     (fun () -> try end_T () with NoMatch -> fake_pmatcher_end ())
   | _ -> 
-    (fun env c _ h -> try let sigma = unify_HO env sigma (EConstr.of_constr c) (EConstr.of_constr t) in EConstr.to_constr sigma (EConstr.of_constr t)
+    (fun env c _ h ->
+       try
+         let sigma = unify_HO env sigma (EConstr.of_constr c) (EConstr.of_constr t) in
+         EConstr.to_constr ~abort_on_undefined_evars:false sigma (EConstr.of_constr t)
```

Comment:
This is highly suspicious code. What are you intending here?

Reply:
I do agree, so I'll try to explain.

Here `(sigma,t)` is an open constr and `(sigma0,c)` is a subterm of the goal, that an open constr. `sigma` extends `sigma0`.
We replace `c` with roughly `nf_evars sigma1 t` where `sigma1` contains the unifier of `t` with `c`.
- If the unifier assigns evars in `c`, then the code is incorrect (since we later work in `sigma0`).
- If the unifier assigns evars in `t` then the code is now correct (up to universe constraints, should be later inferred by `convert_concl` I think).
- If the unifier assigns evars in `t` to terms containing evars (i.e. c is not evar free) then the `~abortion` is triggered for no good reason (since these evars are in `sigma0`).

So, this code from another geologic era (pre-universes) works fine (even today I believe) if the goal contains no evar.
If the goal contains evars and you are lucky it used to work before `~abort` and will work after this PR (it does not work
now, and emilio did not spot it because 99% of the tests on ssreflect work on evar free goals).

I hope I made the misery clear, and that this PR is not making things worse. Does it make sense to you?

Reply:
Why not use Evarutil.nf_evars_universes?

Reply:
Thanks for the hint, I'll look at that.

Reply:
I meant, that I may use this API in another PR to improve this code.
The current PR is just a fix.